generic-ec-core/src/lib.rs

```diff
@@ -185,3 +238,13 @@ impl<N: ArrayLength<u8>> ByteArray for GenericArray<u8, N> {
         GenericArray::default()
     }
 }
+
+/// Reduces an integer represented as array of `N` bytes modulo curve (prime) order
+pub trait Reduce<const N: usize> {
+    /// Interprets `bytes` as big-endian encoding of an integer, returns this
+    /// integer modulo curve (prime) order
+    fn from_be_array_mod_order(bytes: &[u8; N]) -> Self;
+    /// Interprets `bytes` as little-endian encoding of an integer, returns this
+    /// integer modulo curve (prime) order
+    fn from_le_array_mod_order(bytes: &[u8; N]) -> Self;
+}
```

Comment:
This go me to think: why do we have GenericArray in this project at all instead of using const generics? Do you remember what's stopping us?

Reply:
Const generics in the traits is very limited. E.g. this code doesn't compile:

```rust
trait Foo {
    const N: usize;
    fn foo(&self, arr: [u8; Self::N]);
}
```

(which we actually need most of the times)

Reply:
Ok, I actually see one meaningful use of GenericArray, and it's the one that is forbidden similarly to what you show